Known to be broken upstream.

Both openssh-server and openssh-client are installed on the builders,
thus, even if they are not in $checkdepends, these tests are still run
and we need to disable them explicitly.

See: https://gitlab.com/guile-git/guile-git/-/issues/29

diff -upr guile-git-v0.5.2.orig/tests/ssh.scm.in guile-git-v0.5.2/tests/ssh.scm.in
--- guile-git-v0.5.2.orig/tests/ssh.scm.in	2023-08-04 17:52:53.023272203 +0200
+++ guile-git-v0.5.2/tests/ssh.scm.in	2023-08-04 17:53:17.196663353 +0200
@@ -97,8 +97,7 @@ StrictModes no~%"
              (access? shell (logior R_OK X_OK))))))
 
 (define-syntax-rule (with-sshd-server port tests ...)
-  (let ((skip? (or (not (sshd-available?))
-                   (not (valid-user-shell?)))))
+  (let ((skip? #t)) ;; always skip SSH tests, they are broken.
     (dynamic-wind
       (lambda ()
         (if skip?
